packages/react-components/react-motion/library/src/factories/createMotionComponent.ts

```diff
@@ -94,18 +96,26 @@ export function createMotionComponent<MotionParams extends Record<string, Motion
 
       if (element) {
         const atoms = typeof value === 'function' ? value({ element, ...paramsRef.current }) : value;
-        onMotionStart();
 
         const handle = animateAtoms(element, atoms, { isReducedMotion: isReducedMotion() });
 
+        if (!disableMotions) {
+          onMotionStart();
+        }
```

Comment:
I think that it should be as is: otherwise we can break components that already really on callbacks